cli/tools/registry/tar.rs

```diff
@@ -45,75 +44,60 @@ pub fn create_gzipped_tarball(
   unfurler: &SpecifierUnfurler,
   file_patterns: Option<FilePatterns>,
 ) -> Result<PublishableTarball, AnyError> {
+  let file_patterns = file_patterns
+    .unwrap_or_else(|| FilePatterns::new_with_base(dir.to_path_buf()));
   let mut tar = TarGzArchive::new();
   let mut files = vec![];
 
-  let mut paths = HashSet::new();
-
-  let mut ob = OverrideBuilder::new(dir);
-  ob.add("!.git")?.add("!node_modules")?.add("!.DS_Store")?;
-
-  for pattern in file_patterns.as_ref().iter().flat_map(|p| p.include.iter()) {
-    for path_or_pat in pattern.inner() {
-      match path_or_pat {
-        PathOrPattern::Path(p) => ob.add(p.to_str().unwrap())?,
-        PathOrPattern::Pattern(p) => ob.add(p.as_str())?,
-        PathOrPattern::RemoteUrl(_) => continue,
-      };
+  let iter_paths = FileCollector::new(|e| {
+    if !e.file_type.is_file() {
+      if let Ok(specifier) = ModuleSpecifier::from_file_path(e.path) {
```

Comment:
I'm confused here - if the file type is not a file then why try to do `ModuleSpecifier::from_file_path`? Can it be a symlink?